app/src/main/java/com/techlads/composetv/features/domain/paging/NowPlayingMovieSource.kt

```diff
@@ -0,0 +1,32 @@
+package com.techlads.composetv.features.domain.paging
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.techlads.composetv.features.domain.model.Movie
+import com.techlads.composetv.features.domain.repository.MoviesRepository
+import retrofit2.HttpException
+import java.io.IOException
+import javax.inject.Inject
+
+class NowPlayingMovieSource @Inject constructor(private val repository: MoviesRepository):
```

Comment:
Datasource should not consume the Repository, it's the other way around. Movies repository will consume data source and then usecase will consume the repository 

viewmodel -> usecase -> repository -> data source